components/fxa-client/src/lib.rs

```diff
@@ -124,6 +137,27 @@ impl FirefoxAccount {
         Ok(url)
     }
 
+    /// Handle any incoming push message payload coming from the Firefox Accounts
+    /// servers.
+    pub fn handle_push_message(&mut self, payload: &str) -> Result<Vec<AccountEvent>> {
+        let payload = serde_json::from_str(payload)?;
+        match payload {
+            PushPayload::CommandReceived(_) => self.poll_remote_commands(),
+        }
+    }
+
+    fn get_refresh_token(&self) -> Result<&str> {
+        match self.state.refresh_token {
+            Some(ref token_info) => Ok(&token_info.token),
+            None => Err(ErrorKind::NoRefreshToken.into()),
+        }
+    }
+
+    /// Registers a persist callback that will be called automatically every
+    /// time the account state needs to be persisted.
+    /// `FirefoxAccount` can be then restored using the `from_json` method.
+    /// Warning: It is your responsability to persist this information in a secure
+    /// storage as it can contain Sync Keys and OAuth tokens!
```

Comment:
This warning should probably be in the `.kt` wrapper where callers will see it, rather than hidden away in the rust code here.